### ISSUE TITLE
Cursor position moves on scroll e.g. ctrl+d, ctrl+u

### ISSUE THREAD

**Issue by gerardroche** [NeoVintageous/NeoVintageous#242]
<a href="https://github.com/gerardroche"><img src="https://avatars0.githubusercontent.com/u/44148?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [gerardroche](https://github.com/gerardroche)**
_Thursday Jun 25, 2015 at 02:45 GMT_
_Originally opened as https://github.com/guillermooo/Vintageous/issues/979_

----

Currently scrolling moves the cursor position. In Vim the relative cursor position doesn't move.

Try it in Vim and comparing it to ST. The difference is not difficult to see. 

```
CTRL-D            Scroll window Downwards in the buffer.  The number of
            lines comes from the 'scroll' option (default: half a
            screen).  If [count] given, first set 'scroll' option
            to [count].  The cursor is moved the same number of
            lines down in the file (if possible; when lines wrap
            and when hitting the end of the file there may be a
            difference).  When the cursor is on the last line of
            the buffer nothing happens and a beep is produced.
            See also 'startofline' option.
            {difference from vi: Vim scrolls 'scroll' screen
            lines, instead of file lines; makes a difference when
            lines wrap}
```

**Comment by gerardroche** [comment]
<a href="https://github.com/davea1"><img src="https://avatars3.githubusercontent.com/u/6767352?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [davea1](https://github.com/davea1)**
_Friday Sep 18, 2015 at 18:56 GMT_

----

+1

**Comment by gerardroche** [comment]
Docs https://neovim.io/doc/user/scroll.html

**Comment by gerardroche** [comment]
@ davea1 This will be fixed in the next version, 1.8.0 (NeoVintageous is a fork of VIntageous).

**Comment by davea1** [comment]
@gerardroche Awesome. Thanks!

**Comment by StabbarN** [comment]
>  In Vim the relative cursor position doesn't move.

What's the source of this? Is this really an issue?

I'd argue that the cursor should move when scroll using `ctrl-d` and `ctrl-u`. The cursor does move for me when using Neovim v0.3.4 without any plugins. Please revert the change made in 1.8.0.

See gif of me using neovim in terminal. Download link active for 100 downloads and 7 days from now:
https://send.firefox.com/download/83a14ff518/#uBuhkysKfukpeuZReH-VDQ

**Comment by gerardroche** [comment]
I was unable to view the gif. The cursor should stay on the same screen line when scrolling via ctrl-d and ctrl-u. I've tested on a vanilla vim 8.0 `vim -u NONE`. Perhaps there's some configuration or OS specific configuration thing going on? Try `vim -u NONE` and see if it's the same.

**Comment by StabbarN** [comment]
Ah, yes, I see Thanks for explaining. On `vim -u NONE` it stays on same screen line. I've tested more with Sublime (version 3.2) and what's happening is:
1) My screen shows 53 lines and the cursor is on the first line.
2) I scroll down `ctrl-d` three times. Each time moving the screen 40 lines. So the screen jumps from displaying lines 1-53 to 121-173.
3) I tap `j`, and the screen scrolls up to row 82.

The cursor seem to lag one `ctrl-d` behind. If I tap `ctrl-d` 5 times the cursor gets on 5*40+2=162 but the screen shows lines 201-253.

I'm leaning more towards that this problem is caused by Sublime 3.2 and not Neovintageous 1.8. 
I've tried removing my settings in Sublime but to no avail.
